ceph/ceph_admin/helper.py

```diff
@@ -2,8 +2,216 @@
+        return node.shortname
+
+    @staticmethod
+    def get_addr(node):
+        return node.ip_address
+
+    @staticmethod
+    def get_labels(node):
+        return node.role.role_list
+
+    def _get_template(self, service_type):
+        """
+        Return Jinja template based on the service_type
+        Args:
+            service_type: service name (ex., "host")
+        Returns:
+            template
+        """
+        path = self.template_path + f"{service_type}.jinja"
+        with open(path) as fd:
+            template = fd.read()
+        return Template(template)
+
+    def render_hosts(self, spec):
+        """
+        Return hosts spec content based on host config
+        args:
+            spec: hosts specification
+
+        spec:
+          - service_type: host
+            address: true
+            labels: apply-all-labels
+            nodes:
+                - node2
+                - node3
+        Returns:
+            hosts_spec
+        """
+        template = self._get_template("host")
+        hosts = []
+        address = spec.get("address")
+        labels = spec.get("labels")
+        for node_name in spec["nodes"]:
+            host = dict()
+            node = get_node_by_id(self.cluster, node_name)
+            host["hostname"] = self.get_hostname(node)
+            if address:
+                host["address"] = self.get_addr(node)
+            if labels:
+                host["labels"] = self.get_labels(node)
+            hosts.append(host)
+
+        return template.render(hosts=hosts)
+
+    def render_common_services(self, spec):
+        """
+        prepare spec content for common services
+        which is mentioned in COMMON_SERVICES
+         - mon
+         - mgr
+         - alertmanager
+         - crash
+         - grafana
+         - node-exporter
+         - prometheus
+
+        Args:
+            spec: common service spec config
+
+        spec:
+          - service_type: mon
+            unmanaged: boolean    # true or false
+            placement:
+              count: 2
+              label: "mon"
+              host_pattern: "*"   # either hosts or host_pattern
+              nodes:
+                - node2
+                - node3
+        Returns:
+            service_spec
+        """
+        template = self._get_template("common_svc_template")
+        node_names = spec["placement"].pop("nodes", None)
+        if node_names:
+            spec["placement"]["hosts"] = []
+            nodes = get_nodes_by_ids(self.cluster, node_names)
+            for node in nodes:
+                spec["placement"]["hosts"].append(node.shortname)
+
+        return template.render(spec=spec)
+
+    def _get_render_method(self, service_type):
+        """
+        Return render definition based on service_type
+        Args:
+            service_type: service name
+        Returns:
+            method
+        """
+        render_definitions = {
+            "host": self.render_hosts,
+        }
```

Comment:
Would prefer to go with current approach, because this `render_definitions` dict will grow if service is taken individually.
where defining number of same method reference will be reduced and any service needs special attention can be specified in the `render_definitions` like RGW, NFS., etc.
```
[mon|mgr|rbd-mirror|cephfs-mirror|crash|alertmanager|grafana|node-exporter| prometheus|mds|rgw|nfs|iscsi|cephadm-exporter ] 
```